docs/overriding_agg_fn.rst

```diff
@@ -32,12 +32,125 @@ Choose from the following predefined aggregation functions:
 - ``openfl.component.aggregation_functions.WeightedAverage`` (default)
 - ``openfl.component.aggregation_functions.Median``
 - ``openfl.component.aggregation_functions.GeometricMedian``
+- ``openfl.component.aggregation_functions.AdagradAdaptiveAggregation``
+- ``openfl.component.aggregation_functions.AdamAdaptiveAggregation``
+- ``openfl.component.aggregation_functions.YogiAdaptiveAggregation``
 
 
+.. note::
+    To create adaptive aggregation functions,
+    the user must specify parameters for the aggregation optimizer
+    (``NumPyAdagrad``, ``NumPyAdam`` or ``NumPyYogi``) that will aggregate
+    the global model. Theese parameters parameters are passed via **keywords**.
+
+    Also, user must pass one of the arguments: ``params``
+    - model parameters (a dictionary with named model parameters
+    in the form of numpy arrays), or pass ``model_interface``
+    - an instance of the `ModelInterface <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/experiment.py>`_ class.
+    If user pass both ``params`` and ``model_interface``,
+    then the optimizer parameters are initialized via
+    ``params``, ignoring ``model_interface`` argument.
+
+    See the `AdagradAdaptiveAggregation
+    <https://github.com/intel/openfl/blob/develop/openfl/component/aggregation_functions/adagrad_adaptive_aggregation.py>`_
+    definitions for details.
+
+    `Adaptive federated optimization <https://arxiv.org/pdf/2003.00295.pdf>`_ original paper.
+
+``AdagradAdaptiveAggregation`` usage example:
+
+.. code-block:: python
+
+    from openfl.interface.interactive_api.experiment import TaskInterface, ModelInterface
+    from openfl.component.aggregation_functions import AdagradAdaptiveAggregation
+
+    TI = TaskInterface()
+    MI = ModelInterface(model=model,
+                        optimizer=optimizer,
+                        framework_plugin=framework_adapter)
+    ...
+
+    # Creating aggregation function
+    agg_fn = AdagradAdaptiveAggregation(model_interface=MI,
```

Comment:
Is it correct that adaptive aggregation and a custom aggregation function are mutually exclusive?
If we imagine a set of linear functions that merge multiple local updates to a global one (aggregation)
`global_model = L(local_models) = L (global(-1 round) + local updates) = global(-1 round) + L(local updates)`
It seems like the aggregation functions and optimizers may coexist